In [1]:
import pandas as pd
from pandas import DataFrame
import numpy as np
import datetime
from datetime import datetime, timedelta
#import plotly.graph_objs as go
import matplotlib.pyplot as plt
import csv
import seaborn as sns
%matplotlib inline
sns.set()
from PIL import Image

# chik dataset:

In [2]:
import sys
import glob

path = r'/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik'
filenames = glob.glob(path + "/*.csv")
filenames

['/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKNET 2014_16.02.2018.csv',
 '/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKNET 2015_04.05.2018.csv',
 '/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKON 2016_16.10.2018.csv',
 '/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKON 2017_18.07.2018.csv',
 '/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKON 2018_15.10.2018.csv',
 '/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIK_BRASIL_2019.csv']

In [120]:
df1 = pd.read_csv(filenames[5], low_memory = False, encoding="iso-8859-1")

In [121]:
for value in df1.columns:
    print(value)

IMUNOH_N
HOSPITALIZ
DT_INTERNA
UF
MUNICIPIO
HOSPITAL
TPAUTOCTO
COUFINF
COPAISINF
COMUNINF
CLASSI_FIN
CRITERIO
DOENCA_TRA
CLINC_CHIK
EVOLUCAO
DT_OBITO
DT_ENCERRA
ALRM_HIPOT
ALRM_PLAQ
ALRM_VOM
ALRM_SANG
ALRM_HEMAT
ALRM_ABDOM
ALRM_LETAR
ALRM_HEPAT
ALRM_LIQ
DT_ALRM
GRAV_PULSO
GRAV_CONV
GRAV_ENCH
GRAV_INSUF
GRAV_TAQUI
GRAV_EXTRE
GRAV_HIPOT
GRAV_HEMAT
GRAV_MELEN
GRAV_METRO
GRAV_SANG
GRAV_AST
GRAV_MIOC
GRAV_CONSC
GRAV_ORGAO
DT_GRAV
MANI_HEMOR
EPISTAXE
GENGIVO
METRO
PETEQUIAS
HEMATURA
SANGRAM
LACO_N
PLASMATICO
EVIDENCIA
PLAQ_MENOR
CON_FHD
COMPLICA
NU_LOTE_I
TP_SISTEMA
NDUPLIC_N
DT_DIGITA
DT_TRANSUS
DT_TRANSDM
DT_TRANSSM
DT_TRANSRM
DT_TRANSRS
DT_TRANSSE
NU_LOTE_V
NU_LOTE_H
CS_FLXRET
FLXRECEBI
IDENT_MICR
MIGRADO_W
NU_NOTIFIC
TP_NOT
ID_AGRAVO
DT_NOTIFIC
SEM_NOT
NU_ANO
SG_UF_NOT
ID_MUNICIP
ID_UNIDADE
DT_SIN_PRI
SEM_PRI
DT_NASC
NU_IDADE_N
CS_SEXO
CS_GESTANT
CS_RACA
CS_ESCOL_N
SG_UF
ID_MN_RESI
CS_ZONA
ID_PAIS
DT_INVEST
ID_OCUPA_N
FEBRE
MIALGIA
CEFALEIA
EXANTEMA
VOMITO
NAUSEA
DOR_COSTAS
CONJUNTVIT
AR

In [122]:
df1.columns= df1.columns.str.lower()
df1['dt_nasc']

0         19780925
1         19670521
2         19990311
3         19801112
4         19520803
            ...   
180778    19980117
180779    19640130
180780    19700624
180781    19920521
180782    19810812
Name: dt_nasc, Length: 180783, dtype: object

In [116]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180783 entries, 0 to 180782
Columns: 136 entries, imunoh_n to date
dtypes: datetime64[ns](1), float64(77), int64(2), object(56)
memory usage: 187.6+ MB


In [123]:
df1['dt_nasc'] = df1['dt_nasc'].replace('        ', '16771001')

In [124]:
df1['dt_nasc'] = df1['dt_nasc'].fillna(0)

In [126]:
df1['date']= pd.to_datetime(df1['dt_nasc'].astype(str), errors='coerce')

In [127]:
df1['date']

0        1978-09-25
1        1967-05-21
2        1999-03-11
3        1980-11-12
4        1952-08-03
            ...    
180778   1998-01-17
180779   1964-01-30
180780   1970-06-24
180781   1992-05-21
180782   1981-08-12
Name: date, Length: 180783, dtype: datetime64[ns]

In [132]:
from datetime import date 
def calculateAge(born): 
    today = date(2019, 6, 1)
    try:  
        birthday = born.replace(year = today.year) 
  
    # raised when birth date is February 29 
    # and the current year is not a leap year 
    except ValueError:  
        birthday = born.replace(year = today.year, 
                  month = born.month + 1, day = 1) 
  
    if birthday > today: 
        return today.year - born.year - 1
    else: 
        return today.year - born.year 

In [134]:
nu_age = []
for value in df1['date']:
    nu_age.append(calculateAge(value))

In [137]:
df1['nu_idade_n'] = nu_age

In [136]:
lst_df = []
for f in filenames:
    print(f)
    df = pd.read_csv(f, low_memory = False, encoding="iso-8859-1")
    df.columns= df.columns.str.lower()
    lst_df.append(df)

/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKNET 2014_16.02.2018.csv
/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKNET 2015_04.05.2018.csv
/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKON 2016_16.10.2018.csv
/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKON 2017_18.07.2018.csv
/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIKON 2018_15.10.2018.csv
/media/juliane_oliveira/My Passport/Datalake/SINAN/Chikungunya/chik/CHIK_BRASIL_2019.csv


In [138]:
lst_df[5] = df1

In [139]:
df = pd.concat(lst_df,sort=True)

In [140]:
df.columns

Index(['acido_pept', 'alrm_abdom', 'alrm_hemat', 'alrm_hepat', 'alrm_hipot',
       'alrm_letar', 'alrm_liq', 'alrm_plaq', 'alrm_sang', 'alrm_vom',
       ...
       'tel_hosp', 'tp_not', 'tp_sistema', 'tpautocto', 'tpuninot', 'uf',
       'unnamed: 132', 'unnamed: 133', 'unnamed: 134', 'vomito'],
      dtype='object', length=164)

In [144]:
df[df.nu_ano == 2019].nu_idade_n.value_counts()

37.0     1996
36.0     1934
38.0     1879
41.0     1871
40.0     1851
         ... 
108.0       1
109.0       1
100.0       1
98.0        1
105.0       1
Name: nu_idade_n, Length: 110, dtype: int64

## We will create year and month from  variable 'dt_notific' in order to use in our work. (see dengue_case_file for the problematic relating to the variable tp extract the week)

In [ ]:
df['Mycol'] =  df['dt_notific'].apply(pd.to_datetime)

In [ ]:
df['year'] = pd.DatetimeIndex(df['Mycol']).year
df['month'] = pd.DatetimeIndex(df['Mycol']).month

In [ ]:
df.year.value_counts() # number of cases per year

In [ ]:
df.classi_fin.value_counts()

# Construct the dataset of chik with the desired variables

In [141]:
df = df[(df.classi_fin == 13)]

In [145]:
df.classi_fin.value_counts()

13.0    422108
Name: classi_fin, dtype: int64

In [158]:
data = df.filter(['id_mn_resi','nu_ano','cs_sexo','nu_idade_n'])

In [159]:
data = data[(data.nu_idade_n != 'F') & (data.nu_idade_n != 'M')]

In [160]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 422108 entries, 10 to 180764
Data columns (total 4 columns):
id_mn_resi    422105 non-null object
nu_ano        422093 non-null float64
cs_sexo       422107 non-null object
nu_idade_n    422083 non-null object
dtypes: float64(1), object(3)
memory usage: 16.1+ MB


## Process to clean age group and padronize following the age category given in population from IBGE

In [151]:
data.head(1)

id_mn_resi  nu_ano cs_sexo nu_idade_n
10     230440  2014.0       M       4021

In [161]:
data = data.fillna(0)

In [162]:
data.nu_idade_n.unique()

array([4021.0, 4050.0, 4048.0, 4045.0, 4051.0, 4032.0, 4074.0, 4036.0,
       4039.0, 4026.0, 4049.0, 4002.0, 4054.0, 2001.0, 4035.0, 4027.0,
       4029.0, 4055.0, 4034.0, 4022.0, 4017.0, 4072.0, 4005.0, 4059.0,
       4069.0, 3001.0, 4063.0, 4015.0, 4040.0, 4010.0, 4044.0, 4024.0,
       4016.0, 3004.0, 4004.0, 4031.0, 4013.0, 4065.0, 4056.0, 4053.0,
       4087.0, 4028.0, 4052.0, 4088.0, 4033.0, 4023.0, 4058.0, 4047.0,
       4080.0, 4079.0, 4030.0, 4042.0, 4070.0, 4041.0, 4019.0, 4011.0,
       4020.0, 4061.0, 4057.0, 3006.0, 4075.0, 4073.0, 4094.0, 4081.0,
       4060.0, 4046.0, 4012.0, 4018.0, 4014.0, 4043.0, 4025.0, 4071.0,
       4066.0, 4068.0, 4062.0, 4009.0, 3009.0, 3007.0, 4089.0, 3003.0,
       4064.0, 4001.0, 2025.0, 4038.0, 4007.0, 2009.0, 4037.0, 4008.0,
       4078.0, 4083.0, 4086.0, 3002.0, 4096.0, 2013.0, 4076.0, 4084.0,
       3005.0, 2020.0, 4067.0, 4006.0, 4003.0, 4092.0, '4014', '4048',
       '4036', '4051', '4071', '4034', '4023', '4031', '4015', '4030',
      

1. Hora
2. Dia
3. Mês
4. Ano
Ex: 3009 – nove meses, 4018 –
dezoito anos

In [163]:
vector =[]
for value in data.nu_idade_n.astype(int).astype(str):
    if value == '1': value = 2001
    elif value == '2': value = 2001
    elif value == '3': value = 3001
    elif 4<= int(value) < 10: value = '400'+value
    elif len(value) == 2: value = '40' + value
    elif len(value) == 3: value = '4' + value
    vector.extend([value])

In [164]:
np.unique(vector)

array(['0', '1000', '1019', '1020', '1029', '1035', '1037', '1057',
       '1066', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015',
       '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023',
       '2024', '2025', '2026', '2027', '2028', '2029', '2030', '2032',
       '2043', '2053', '2058', '2075', '3001', '3002', '3003', '3004',
       '3005', '3006', '3007', '3008', '3009', '3010', '3011', '3021',
       '3031', '3044', '3048', '3055', '40-1', '4000', '4001', '4002',
       '4003', '4004', '4005', '4006', '4007', '4008', '4009', '4010',
       '4011', '4012', '4013', '4014', '4015', '4016', '4017', '4018',
       '4019', '4020', '4021', '4022', '4023', '4024', '4025', '4026',
       '4027', '4028', '4029', '4030', '4031', '4032', '4033', '4034',
       '4035', '4036', '4037', '4038', '4039', '4040', '4041', '4042',
       '4043', '4044', '4045', '4046', '4047', '4048', '4049', '4050',
       '4

In [165]:
data['nu_idade_n_cor'] = vector

In [174]:
data = data[(data.nu_idade_n_cor != '0')& (data.nu_idade_n_cor !='40-1')]

In [175]:
data.columns

Index(['id_mn_resi', 'nu_ano', 'cs_sexo', 'nu_idade_n', 'nu_idade_n_cor'], dtype='object')

In [179]:
data.nu_idade_n_cor.unique()

array(['4021', '4050', '4048', '4045', '4051', '4032', '4074', '4036',
       '4039', '4026', '4049', '4002', '4054', '2001', '4035', '4027',
       '4029', '4055', '4034', '4022', '4017', '4072', '4005', '4059',
       '4069', '3001', '4063', '4015', '4040', '4010', '4044', '4024',
       '4016', '3004', '4004', '4031', '4013', '4065', '4056', '4053',
       '4087', '4028', '4052', '4088', '4033', '4023', '4058', '4047',
       '4080', '4079', '4030', '4042', '4070', '4041', '4019', '4011',
       '4020', '4061', '4057', '3006', '4075', '4073', '4094', '4081',
       '4060', '4046', '4012', '4018', '4014', '4043', '4025', '4071',
       '4066', '4068', '4062', '4009', '3009', '3007', '4089', '3003',
       '4064', '4001', '2025', '4038', '4007', '2009', '4037', '4008',
       '4078', '4083', '4086', '3002', '4096', '2013', '4076', '4084',
       '3005', '2020', '4067', '4006', '4003', '4092', '4085', '3010',
       '4077', '2029', '2022', '4090', '4082', '3008', '4098', '4093',
      

## Categorize age group
### 0,  101,  202,  303,  404,  505,  606,  707,  808,  909, 1010,
###       1111, 1212, 1313, 1414, 1515, 1616, 1717, 1818, 1919, 2024, 2529,
###       3034, 3539, 4044, 4549, 5054, 5559, 6064, 6569, 7074, 7579, 8099],

In [180]:
fx = []
for value in data.nu_idade_n_cor:
    if 1000 <= int(value) <4005: faixa = '4' 
    elif 4005 <=int(value)< 4010: faixa = '509'
    elif 4010 <=int(value)< 4015: faixa = '1014'
    elif 4015 <=int(value)< 4020: faixa = '1519'
    elif 4020 <=int(value)< 4025: faixa = '2024'
    elif 4025 <=int(value)< 4030: faixa = '2529'
    elif 4030 <=int(value)< 4035: faixa = '3034'
    elif 4035 <=int(value)< 4040: faixa = '3539' 
    elif 4040 <=int(value)< 4045: faixa = '4044'   
    elif 4045 <=int(value)< 4050: faixa = '4549'
    elif 4050 <=int(value)< 4055: faixa = '5054'
    elif 4055 <=int(value)< 4060: faixa = '5559'
    elif 4060 <=int(value)< 4065: faixa = '6064'
    elif 4065 <=int(value)< 4070: faixa = '6569'
    elif 4070 <=int(value)< 4075: faixa = '7074'
    elif 4075 <=int(value)< 4080: faixa = '7579'
    elif 4080 <=int(value)< 8100: faixa = '8099'
    fx.extend([faixa])             

In [181]:
data['faixa'] = fx
data

id_mn_resi  nu_ano cs_sexo nu_idade_n nu_idade_n_cor faixa
10         230440  2014.0       M       4021           4021  2024
2161       291080  2014.0       F       4050           4050  5054
3989       510340  2014.0       F       4048           4048  4549
4239       260510  2014.0       F       4045           4045  4549
4240       261160  2014.0       F       4051           4051  5054
...           ...     ...     ...        ...            ...   ...
180613     172120  2019.0       M         16           4016  1519
180708     170740  2019.0       F         31           4031  3034
180737     172100  2019.0       F          8           4008   509
180756     170820  2019.0       F         17           4017  1519
180764     171488  2019.0       F         33           4033  3034

[421283 rows x 6 columns]

In [182]:
data.cs_sexo.value_counts()

F    261814
M    159127
I       342
Name: cs_sexo, dtype: int64

In [183]:
data['count'] = 1

In [184]:
data= data.rename(columns={"id_mn_resi": "codmunres"})

In [185]:
data= data.rename(columns={"nu_ano": "year"})

In [186]:
dados = data.groupby(['codmunres','year','cs_sexo','faixa'])['count'].sum().reset_index()

In [187]:
dados.cs_sexo = dados.cs_sexo.replace('F', 1)
dados.cs_sexo= dados.cs_sexo.replace('M', 2)

In [190]:
dados.head()

codmunres    year cs_sexo faixa  count
2    110002  2017.0       1     4      2
3    110002  2017.0       1  4549      1
4    110002  2017.0       2  7579      1
5    110002  2018.0       1  1519      1
6    110002  2018.0       1  2529      1

In [189]:
dados = dados[dados.codmunres != 0]

In [192]:
dados = dados[dados.year != 0]

In [193]:
dados.groupby(['year'])['count'].sum()

year
2014.0        10
2015.0       670
2016.0     89530
2017.0    166344
2018.0     60587
2019.0    104125
Name: count, dtype: int64

In [195]:
df.nu_ano.value_counts()

2017.0        166345
2019.0        104924
2016.0         89531
2018.0         60587
2015.0           670
33.0              15
2014.0            10
24.0               9
201918.0           1
20190225.0         1
Name: nu_ano, dtype: int64

# Population

In [221]:
pop = pd.read_csv('/media/juliane_oliveira/My Passport/Datalake/POPULACAO/csv/POPBR12.csv')

In [222]:
pop.columns

Index(['MUNIC_RES', 'ANO', 'SEXO', 'SITUACAO', 'FXETARIA', 'POPULACAO'], dtype='object')

In [223]:
pop.FXETARIA.unique()

array([   0,  101,  202,  303,  404,  505,  606,  707,  808,  909, 1010,
       1111, 1212, 1313, 1414, 1515, 1616, 1717, 1818, 1919, 2024, 2529,
       3034, 3539, 4044, 4549, 5054, 5559, 6064, 6569, 7074, 7579, 8099])

In [224]:
pop = pop.replace({'FXETARIA': {0: 4, 101:4, 202:4,  303:4,  404:4,
                                505: 509, 606: 509,  707: 509,  808: 509,  909: 509,
                                1010: 1014, 1111: 1014, 1212: 1014, 1313: 1014, 1414: 1014,
                                1515: 1519, 1616: 1519, 1717: 1519, 1818: 1519, 1919: 1519}})

In [225]:
pop.head()

MUNIC_RES   ANO  SEXO  SITUACAO  FXETARIA  POPULACAO
0     110001  2012     1         3         4        187
1     110001  2012     1         3         4        186
2     110001  2012     1         3         4        187
3     110001  2012     1         3         4        190
4     110001  2012     1         3         4        193

# Creating tables to map the standard incidence ratio of chik per year

## expected number of cases per year and cummulated

In [202]:
teste = dados

In [203]:
teste = teste[(2014 <= teste.year) & (teste.year <= 2019) & (teste.cs_sexo != 'f')&(teste.cs_sexo != 'I')]

In [204]:
teste.cs_sexo.unique()

array([1, 2], dtype=object)

In [205]:
teste.year.unique()

array([2017., 2018., 2016., 2015., 2014., 2019.])

In [206]:
teste['faixa'] = teste['faixa'].astype(int)

/home/juliane_oliveira/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [207]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 35930 entries, 2 to 36208
Data columns (total 5 columns):
codmunres    35930 non-null object
year         35930 non-null float64
cs_sexo      35930 non-null object
faixa        35930 non-null int64
count        35930 non-null int64
dtypes: float64(1), int64(2), object(2)
memory usage: 1.6+ MB


In [261]:
sum_observed_gender_age_year = []
for year in [2014,2015,2016,2017,2018,2019]:
    df_year = teste[teste.year == year]
    table_year = pd.pivot_table(df_year, values='count', index=['codmunres'], columns=['cs_sexo','faixa'])
    table_year=table_year.fillna(0)
    resulted_dataframe = table_year.sum().reset_index()
    sum_observed_gender_age_year.append(resulted_dataframe)

In [263]:
sum_observed_gender_age_year[5]

cs_sexo  faixa       0
0         1      4   597.0
1         1    509  1427.0
2         1   1014  2129.0
3         1   1519  3427.0
4         1   2024  4836.0
5         1   2529  4908.0
6         1   3034  5364.0
7         1   3539  5956.0
8         1   4044  5709.0
9         1   4549  5549.0
10        1   5054  5550.0
11        1   5559  5070.0
12        1   6064  4476.0
13        1   6569  3362.0
14        1   7074  2157.0
15        1   7579  1325.0
16        1   8099  2283.0
17        2      4   728.0
18        2    509  1539.0
19        2   1014  2376.0
20        2   1519  2550.0
21        2   2024  3410.0
22        2   2529  3208.0
23        2   3034  3260.0
24        2   3539  3443.0
25        2   4044  3155.0
26        2   4549  3044.0
27        2   5054  2849.0
28        2   5559  2704.0
29        2   6064  2374.0
30        2   6569  1893.0
31        2   7074  1248.0
32        2   7579   801.0
33        2   8099  1298.0

In [212]:
pop.head()

MUNIC_RES   ANO  SEXO  FXETARIA  POPULACAO
0     110001  2000     1         4        118
1     110001  2000     1         4        110
2     110001  2000     1         4        127
3     110001  2000     1         4        149
4     110001  2000     1         4        140

# Population 

In [226]:
pop = pop.astype(int)

In [227]:
teste_pop = pop.groupby(['MUNIC_RES','ANO','SEXO','FXETARIA'])['POPULACAO'].sum().reset_index()
teste_pop.head()

MUNIC_RES   ANO  SEXO  FXETARIA  POPULACAO
0     110001  2012     1         4        943
1     110001  2012     1       509       1058
2     110001  2012     1      1014       1239
3     110001  2012     1      1519       1343
4     110001  2012     1      2024       1090

In [228]:
table_pop = pd.pivot_table(teste_pop, values='POPULACAO', index=['MUNIC_RES'], columns=['SEXO','FXETARIA'])
table_pop

SEXO           1                                                          \
FXETARIA    4       509     1014    1519    2024    2529    3034    3539   
MUNIC_RES                                                                  
110001       943    1058    1239    1343    1090    1039     938     866   
110002      3742    4288    4865    4598    4574    4271    3854    3522   
110003       234     246     304     307     220     243     230     235   
110004      2987    3234    3792    4016    3833    3483    3201    2966   
110005       620     694     831     865     757     674     632     581   
...          ...     ...     ...     ...     ...     ...     ...     ...   
522200       423     524     577     557     530     536     566     521   
522205       297     314     333     338     386     363     358     338   
522220       243     266     249     229     242     370     263     194   
522230       186     240     274     236     200     173     208     213   
530010     98864  104928  113968  112108  124044  131400  122121  103098   

SEXO                     ...       2                                      \
FXETARIA    4044   4549  ...    3539    4044   4549   5054   5559   6064   
MUNIC_RES                ...                                               
110001       901    835  ...     905     792    710    508    425    307   
110002      3378   2682  ...    3606    3208   2494   1877   1523   1069   
110003       240    207  ...     209     213    196    168    138     91   
110004      2842   2459  ...    3159    2859   2420   1963   1441   1018   
110005       632    575  ...     655     623    542    402    307    298   
...          ...    ...  ...     ...     ...    ...    ...    ...    ...   
522200       506    410  ...     522     458    416    308    234    209   
522205       304    272  ...     286     301    217    176    157     82   
522220       158    139  ...     138     148    109     81     62     38   
522230       192    173  ...     163     183    160    121    136     94   
530010     91493  74737  ...  116108  101992  87146  69220  54147  40270   

SEXO                                   
FXETARIA    6569   7074   7579   8099  
MUNIC_RES                              
110001       261    175    117     72  
110002       755    510    325    302  
110003        63     66     33     26  
110004       803    613    435    383  
110005       194    133     96     83  
...          ...    ...    ...    ...  
522200       172    142     80     99  
522205        71     56     27     37  
522220        44     14     18     12  
522230        63     44     31     23  
530010     27736  20499  12790  14770  

[5565 rows x 34 columns]

In [229]:
table_pop=table_pop.fillna(0)

sum_pop_gender_age_year = table_pop.sum().reset_index()
sum_pop_gender_age_year

SEXO  FXETARIA        0
0      1         4  7143389
1      1       509  7758474
2      1      1014  8875967
3      1      1519  8704999
4      1      2024  8782606
5      1      2529  8610952
6      1      3034  7853458
7      1      3539  6883541
8      1      4044  6427350
9      1      4549  5785459
10     1      5054  4912910
11     1      5559  3963556
12     1      6064  3087312
13     1      6569  2256975
14     1      7074  1691406
15     1      7579  1105881
16     1      8099  1148647
17     2         4  6901204
18     2       509  7474673
19     2      1014  8587202
20     2      1519  8577046
21     2      2024  8766636
22     2      2529  8795610
23     2      3034  8166586
24     2      3539  7243527
25     2      4044  6800153
26     2      4549  6240934
27     2      5054  5389329
28     2      5559  4441120
29     2      6064  3520063
30     2      6569  2654866
31     2      7074  2103802
32     2      7579  1493580
33     2      8099  1827317

In [215]:
#sum_pop_gender_age_year = []
#for year in [2014,2015,2016,2017,2019]:
#    df_year = teste_pop[teste_pop.ANO == year]
#    table_year = pd.pivot_table(df_year, values='POPULACAO', index=['MUNIC_RES'], columns=['SEXO','FXETARIA'])
#    table_year=table_year.fillna(0)
#    resulted_dataframe = table_year.sum().reset_index()
#    sum_pop_gender_age_year.append(resulted_dataframe)

#sum_pop_gender_age_year[2]

# calculating r_ j (for each year)

In [ ]:
#r_j = []
#for df1,df2 in zip(sum_observed_gender_age_year,sum_pop_gender_age_year):
    #df2.rename(columns={0:'pop'}, inplace=True)
#    risco_relativo = df1[0]/df2[0]
#    risco_relativo = risco_relativo.fillna(0)
#    r_j.append(risco_relativo)

In [264]:
r_j = []
for df1 in sum_observed_gender_age_year:
    #df2.rename(columns={0:'pop'}, inplace=True)
    risco_relativo = df1[0]/sum_pop_gender_age_year[0]
    risco_relativo = risco_relativo.fillna(0)
    r_j.append(risco_relativo)

In [265]:
r_j[5]

0     0.000084
1     0.000184
2     0.000240
3     0.000394
4     0.000551
5     0.000570
6     0.000683
7     0.000865
8     0.000888
9     0.000959
10    0.001130
11    0.001279
12    0.001450
13    0.001490
14    0.001275
15    0.001198
16    0.001988
17    0.000105
18    0.000206
19    0.000277
20    0.000297
21    0.000389
22    0.000365
23    0.000399
24    0.000475
25    0.000464
26    0.000488
27    0.000529
28    0.000609
29    0.000674
30    0.000713
31    0.000593
32    0.000536
33    0.000710
Name: 0, dtype: float64

# calculating E_i for each municipality

# 2014

In [266]:
#filter_2014 = teste_pop[teste_pop.ANO == 2014]
#P_2014 = pd.pivot_table(filter_2014, values='POPULACAO', index=['MUNIC_RES'], columns=['SEXO','FXETARIA'])
#P_2014;

In [267]:
P_2014 = table_pop

In [268]:
E_2014 =[]
for i in range(len(P_2014)):
    E_i = sum(np.array(P_2014.iloc[i])*np.array(r_j[0]))
    E_2014.append([P_2014.index.values[i],E_i])

In [269]:
Expected_2014 = pd.DataFrame(E_2014,
                 columns=['codmunres', 'expected_2014'])

In [270]:
Expected_2014.head()

codmunres  expected_2014
0     110001       0.001357
1     110002       0.005289
2     110003       0.000318
3     110004       0.004277
4     110005       0.000897

# 2015

In [271]:
#filter_2015 = teste_pop[teste_pop.ANO == 2015]
#P_2015 = pd.pivot_table(filter_2015, values='POPULACAO', index=['MUNIC_RES'], columns=['SEXO','FXETARIA'])
#P_2015;

In [272]:
P_2015 = table_pop

In [273]:
E_2015 =[]
for i in range(len(P_2015)):
    E_i = sum(np.array(P_2015.iloc[i])*np.array(r_j[1]))
    E_2015.append([P_2015.index.values[i],E_i])

In [274]:
Expected_2015 = pd.DataFrame(E_2015,
                 columns=['codmunres', 'expected_2015'])

In [275]:
Expected_2015.head()

codmunres  expected_2015
0     110001       0.084080
1     110002       0.316840
2     110003       0.021820
3     110004       0.273317
4     110005       0.058568

# 2016

In [276]:
#filter_2016 = teste_pop[teste_pop.ANO == 2016]
#P_2016 = pd.pivot_table(filter_2016, values='POPULACAO', index=['MUNIC_RES'], columns=['SEXO','FXETARIA'])
#P_2016;

In [277]:
P_2016 = table_pop

In [278]:
E_2016 =[]
for i in range(len(P_2016)):
    E_i = sum(np.array(P_2016.iloc[i])*np.array(r_j[2]))
    E_2016.append([P_2016.index.values[i],E_i])

In [279]:
Expected_2016 = pd.DataFrame(E_2016,
                 columns=['codmunres', 'expected_2016'])

In [280]:
Expected_2016.head()

codmunres  expected_2016
0     110001      11.047781
1     110002      41.231349
2     110003       2.903118
3     110004      36.090898
4     110005       7.776553

# 2017

In [281]:
#filter_2017 = teste_pop[teste_pop.ANO == 2017]
#P_2017 = pd.pivot_table(filter_2017, values='POPULACAO', index=['MUNIC_RES'], columns=['SEXO','FXETARIA'])
#P_2017;

In [282]:
P_2017 = table_pop

In [283]:
E_2017 =[]
for i in range(len(P_2017)):
    E_i = sum(np.array(P_2017.iloc[i])*np.array(r_j[3]))
    E_2017.append([P_2017.index.values[i],E_i])

In [284]:
Expected_2017 = pd.DataFrame(E_2017,
                 columns=['codmunres', 'expected_2017'])

In [285]:
Expected_2017.head()

codmunres  expected_2017
0     110001      20.593672
1     110002      77.271117
2     110003       5.370519
3     110004      67.413771
4     110005      14.481723

# 2018

In [286]:
P_2018 = table_pop

In [287]:
E_2018 =[]
for i in range(len(P_2018)):
    E_i = sum(np.array(P_2018.iloc[i])*np.array(r_j[4]))
    E_2018.append([P_2018.index.values[i],E_i])

In [288]:
Expected_2018 = pd.DataFrame(E_2018,
                 columns=['codmunres', 'expected_2018'])

In [289]:
Expected_2018.head()

codmunres  expected_2018
0     110001       7.461287
1     110002      27.668192
2     110003       1.977879
3     110004      24.303019
4     110005       5.265250

# 2019

In [290]:
P_2019 = table_pop

In [291]:
E_2019 =[]
for i in range(len(P_2019)):
    E_i = sum(np.array(P_2019.iloc[i])*np.array(r_j[5]))
    E_2019.append([P_2019.index.values[i],E_i])

In [292]:
Expected_2019 = pd.DataFrame(E_2019,
                 columns=['codmunres', 'expected_2019'])

In [293]:
Expected_2019.head()

codmunres  expected_2019
0     110001      12.643632
1     110002      46.704364
2     110003       3.373909
3     110004      41.406791
4     110005       8.994050

# Observed number of cases per municipality and year

In [294]:
teste = teste.astype(int)

In [302]:
teste;

In [295]:
observed = teste.groupby(['codmunres','year'])['count'].sum().reset_index()
observed.head()

codmunres  year  count
0     110002  2017      4
1     110002  2018      5
2     110004  2016      4
3     110004  2017      4
4     110004  2018      8

In [296]:
#observed = observed.astype(int)
observed.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4988 entries, 0 to 4987
Data columns (total 3 columns):
codmunres    4988 non-null int64
year         4988 non-null int64
count        4988 non-null int64
dtypes: int64(3)
memory usage: 117.0 KB


In [297]:
observed.describe()

codmunres         year         count
count    4988.000000  4988.000000   4988.000000
mean   282897.312149  2017.379711     84.387330
std     85466.402191     1.173604   1127.213459
min    110002.000000  2014.000000      1.000000
25%    230720.000000  2016.000000      1.000000
50%    270200.000000  2017.000000      2.000000
75%    320460.000000  2018.000000     10.000000
max    530010.000000  2019.000000  62197.000000

In [298]:
observed[observed.year == 2015]

codmunres  year  count
24       110020  2015      1
178      150140  2015      1
340      150715  2015      1
516      172120  2015      1
559      210140  2015      1
...         ...   ...    ...
4723     500270  2015     36
4827     510340  2015      1
4926     510840  2015      1
4935     520140  2015      2
4961     521380  2015      1

[86 rows x 3 columns]

In [303]:
observed = observed[(2014 <= observed.year) & (observed.year <= 2019)]

In [304]:
observed = pd.pivot_table(observed, values='count', index=['codmunres'], columns=['year'])

In [305]:
observed = observed.fillna(0) 
observed

year       2014  2015  2016  2017  2018  2019
codmunres                                    
110002      0.0   0.0   0.0   4.0   5.0   0.0
110004      0.0   0.0   4.0   4.0   8.0   9.0
110005      0.0   0.0   0.0   2.0   1.0   3.0
110006      0.0   0.0   0.0   0.0   1.0   0.0
110007      0.0   0.0   0.0   0.0   0.0   1.0
...         ...   ...   ...   ...   ...   ...
522010      0.0   0.0   0.0   2.0   1.0   0.0
522140      0.0   0.0   1.0   2.0   0.0   0.0
522160      0.0   0.0   0.0   2.0   0.0   0.0
522185      0.0   0.0   1.0   0.0   2.0   0.0
530010      0.0   0.0  63.0  44.0   8.0  41.0

[2493 rows x 6 columns]

In [311]:
observed[2019].sum()

104005.0

In [308]:
df.nu_ano.value_counts()

2017.0        166345
2019.0        104924
2016.0         89531
2018.0         60587
2015.0           670
33.0              15
2014.0            10
24.0               9
201918.0           1
20190225.0         1
Name: nu_ano, dtype: int64

# calculate the standard incidence ratio

In [312]:
codigo_muni = pd.read_excel('/media/juliane_oliveira/My Passport/Datalake/POPULACAO/População/POP_python/DTB_BRASIL_MUNICIPIO.xls')

In [313]:
codigo_muni.head()

UF   Nome_UF  Mesorregião Geográfica   Nome_Mesorregião  \
0  11  Rondônia                       2  Leste Rondoniense   
1  11  Rondônia                       2  Leste Rondoniense   
2  11  Rondônia                       2  Leste Rondoniense   
3  11  Rondônia                       2  Leste Rondoniense   
4  11  Rondônia                       2  Leste Rondoniense   

   Microrregião Geográfica Nome_Microrregião  Município  \
0                        6            Cacoal         15   
1                        6            Cacoal        379   
2                        3         Ariquemes        403   
3                        5  Alvorada D'Oeste        346   
4                        3         Ariquemes         23   

   Código Município Completo           Nome_Município  
0                    1100015    Alta Floresta D'Oeste  
1                    1100379  Alto Alegre dos Parecis  
2                    1100403             Alto Paraíso  
3                    1100346         Alvorada D'Oeste  
4                    1100023                Ariquemes

In [314]:
codigo_muni.columns

Index(['UF', 'Nome_UF', 'Mesorregião Geográfica', 'Nome_Mesorregião',
       'Microrregião Geográfica', 'Nome_Microrregião', 'Município',
       'Código Município Completo', 'Nome_Município'],
      dtype='object')

In [315]:
codigo_muni.rename(columns={'Código Município Completo':'codmuni_full'}, 
                 inplace=True);

In [316]:
codigo_muni['codmuni'] = codigo_muni.codmuni_full.floordiv(10)

In [317]:
codigo_muni.head()

UF   Nome_UF  Mesorregião Geográfica   Nome_Mesorregião  \
0  11  Rondônia                       2  Leste Rondoniense   
1  11  Rondônia                       2  Leste Rondoniense   
2  11  Rondônia                       2  Leste Rondoniense   
3  11  Rondônia                       2  Leste Rondoniense   
4  11  Rondônia                       2  Leste Rondoniense   

   Microrregião Geográfica Nome_Microrregião  Município  codmuni_full  \
0                        6            Cacoal         15       1100015   
1                        6            Cacoal        379       1100379   
2                        3         Ariquemes        403       1100403   
3                        5  Alvorada D'Oeste        346       1100346   
4                        3         Ariquemes         23       1100023   

            Nome_Município  codmuni  
0    Alta Floresta D'Oeste   110001  
1  Alto Alegre dos Parecis   110037  
2             Alto Paraíso   110040  
3         Alvorada D'Oeste   110034  
4                Ariquemes   110002

In [318]:
observed = observed.astype(int)
observed.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2493 entries, 110002 to 530010
Data columns (total 6 columns):
2014    2493 non-null int64
2015    2493 non-null int64
2016    2493 non-null int64
2017    2493 non-null int64
2018    2493 non-null int64
2019    2493 non-null int64
dtypes: int64(6)
memory usage: 136.3 KB


In [319]:
final = codigo_muni.set_index('codmuni').join(observed) 

In [320]:
final

UF           Nome_UF  Mesorregião Geográfica   Nome_Mesorregião  \
codmuni                                                                    
110001   11          Rondônia                       2  Leste Rondoniense   
110037   11          Rondônia                       2  Leste Rondoniense   
110040   11          Rondônia                       2  Leste Rondoniense   
110034   11          Rondônia                       2  Leste Rondoniense   
110002   11          Rondônia                       2  Leste Rondoniense   
...      ..               ...                     ...                ...   
522200   52             Goiás                       5         Sul Goiano   
522205   52             Goiás                       5         Sul Goiano   
522220   52             Goiás                       4       Leste Goiano   
522230   52             Goiás                       4       Leste Goiano   
530010   53  Distrito Federal                       1   Distrito Federal   

         Microrregião Geográfica    Nome_Microrregião  Município  \
codmuni                                                            
110001                         6               Cacoal         15   
110037                         6               Cacoal        379   
110040                         3            Ariquemes        403   
110034                         5     Alvorada D'Oeste        346   
110002                         3            Ariquemes         23   
...                          ...                  ...        ...   
522200                        16         Pires do Rio      22005   
522205                        15           Meia Ponte      22054   
522220                        12  Entorno de Brasília      22203   
522230                        12  Entorno de Brasília      22302   
530010                         1             Brasília        108   

         codmuni_full           Nome_Município  2014  2015  2016  2017  2018  \
codmuni                                                                        
110001        1100015    Alta Floresta D'Oeste   NaN   NaN   NaN   NaN   NaN   
110037        1100379  Alto Alegre dos Parecis   0.0   0.0   1.0   0.0   2.0   
110040        1100403             Alto Paraíso   0.0   0.0   1.0   1.0   1.0   
110034        1100346         Alvorada D'Oeste   NaN   NaN   NaN   NaN   NaN   
110002        1100023                Ariquemes   0.0   0.0   0.0   4.0   5.0   
...               ...                      ...   ...   ...   ...   ...   ...   
522200        5222005               Vianópolis   NaN   NaN   NaN   NaN   NaN   
522205        5222054           Vicentinópolis   NaN   NaN   NaN   NaN   NaN   
522220        5222203                 Vila Boa   NaN   NaN   NaN   NaN   NaN   
522230        5222302            Vila Propício   NaN   NaN   NaN   NaN   NaN   
530010        5300108                 Brasília   0.0   0.0  63.0  44.0   8.0   

         2019  
codmuni        
110001    NaN  
110037    1.0  
110040    2.0  
110034    NaN  
110002    0.0  
...       ...  
522200    NaN  
522205    NaN  
522220    NaN  
522230    NaN  
530010   41.0  

[5570 rows x 15 columns]

In [321]:
final = final.join(Expected_2014.set_index('codmunres')).join(Expected_2015.set_index('codmunres')).join(Expected_2016.set_index('codmunres')).join(Expected_2017.set_index('codmunres')).join(Expected_2018.set_index('codmunres')).join(Expected_2019.set_index('codmunres'))

In [322]:
final.columns

Index([                     'UF',                 'Nome_UF',
        'Mesorregião Geográfica',        'Nome_Mesorregião',
       'Microrregião Geográfica',       'Nome_Microrregião',
                     'Município',            'codmuni_full',
                'Nome_Município',                      2014,
                            2015,                      2016,
                            2017,                      2018,
                            2019,           'expected_2014',
                 'expected_2015',           'expected_2016',
                 'expected_2017',           'expected_2018',
                 'expected_2019'],
      dtype='object')

In [323]:
final['SIR_2014'] = final[2014]/final['expected_2014']
final['SIR_2015'] = final[2015]/final['expected_2015']
final['SIR_2016'] = final[2016]/final['expected_2016']
final['SIR_2017'] = final[2017]/final['expected_2017']
final['SIR_2018'] = final[2018]/final['expected_2018']
final['SIR_2019'] = final[2019]/final['expected_2019']

In [324]:
final = final.fillna(0)

In [325]:
final.describe()

UF  Mesorregião Geográfica  Microrregião Geográfica  \
count  5570.000000             5570.000000              5570.000000   
mean     32.377738                4.264991                17.633573   
std       9.833862                3.206175                15.514306   
min      11.000000                1.000000                 1.000000   
25%      25.000000                2.000000                 6.000000   
50%      31.000000                3.000000                13.000000   
75%      41.000000                6.000000                24.000000   
max      53.000000               15.000000                66.000000   

          Município  codmuni_full         2014         2015          2016  \
count   5570.000000  5.570000e+03  5570.000000  5570.000000   5570.000000   
mean   15816.982585  3.253591e+06     0.001795     0.120287     16.063016   
std    15997.299780  9.849103e+05     0.084731     2.409333    262.224073   
min       13.000000  1.100015e+06     0.000000     0.000000      0.000000   
25%     4507.250000  2.512126e+06     0.000000     0.000000      0.000000   
50%    10400.500000  3.146280e+06     0.000000     0.000000      0.000000   
75%    20853.000000  4.119190e+06     0.000000     0.000000      0.000000   
max    72202.000000  5.300108e+06     6.000000   110.000000  16616.000000   

               2017         2018  ...  expected_2016  expected_2017  \
count   5570.000000  5570.000000  ...    5570.000000    5570.000000   
mean      29.844704    10.867684  ...      16.063016      29.844704   
std      854.390073   222.831997  ...      96.648099     179.103530   
min        0.000000     0.000000  ...       0.000000       0.000000   
25%        0.000000     0.000000  ...       2.488363       4.599610   
50%        0.000000     0.000000  ...       5.139014       9.525486   
75%        0.000000     0.000000  ...      10.807899      20.174028   
max    62197.000000  9800.000000  ...    5359.407640    9928.589064   

       expected_2018  expected_2019     SIR_2014     SIR_2015     SIR_2016  \
count    5570.000000    5570.000000  5570.000000  5570.000000  5570.000000   
mean       10.867684      18.672352     0.126678     0.449231     0.647067   
std        65.602622     113.493686     7.445208    11.539818     4.146955   
min         0.000000       0.000000     0.000000     0.000000     0.000000   
25%         1.685108       2.909676     0.000000     0.000000     0.000000   
50%         3.475139       5.977951     0.000000     0.000000     0.000000   
75%         7.292242      12.492703     0.000000     0.000000     0.000000   
max      3637.868685    6295.465040   547.379896   785.154514   107.070305   

          SIR_2017     SIR_2018     SIR_2019  
count  5570.000000  5570.000000  5570.000000  
mean      0.481444     0.269722     0.283515  
std       2.874615     3.017699     2.432193  
min       0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000  
50%       0.000000     0.000000     0.000000  
75%       0.000000     0.000000     0.000000  
max      73.538175   126.134742    76.880912  

[8 rows x 23 columns]

In [326]:
final

UF           Nome_UF  Mesorregião Geográfica   Nome_Mesorregião  \
codmuni                                                                    
110001   11          Rondônia                       2  Leste Rondoniense   
110037   11          Rondônia                       2  Leste Rondoniense   
110040   11          Rondônia                       2  Leste Rondoniense   
110034   11          Rondônia                       2  Leste Rondoniense   
110002   11          Rondônia                       2  Leste Rondoniense   
...      ..               ...                     ...                ...   
522200   52             Goiás                       5         Sul Goiano   
522205   52             Goiás                       5         Sul Goiano   
522220   52             Goiás                       4       Leste Goiano   
522230   52             Goiás                       4       Leste Goiano   
530010   53  Distrito Federal                       1   Distrito Federal   

         Microrregião Geográfica    Nome_Microrregião  Município  \
codmuni                                                            
110001                         6               Cacoal         15   
110037                         6               Cacoal        379   
110040                         3            Ariquemes        403   
110034                         5     Alvorada D'Oeste        346   
110002                         3            Ariquemes         23   
...                          ...                  ...        ...   
522200                        16         Pires do Rio      22005   
522205                        15           Meia Ponte      22054   
522220                        12  Entorno de Brasília      22203   
522230                        12  Entorno de Brasília      22302   
530010                         1             Brasília        108   

         codmuni_full           Nome_Município  2014  ...  expected_2016  \
codmuni                                               ...                  
110001        1100015    Alta Floresta D'Oeste   0.0  ...      11.047781   
110037        1100379  Alto Alegre dos Parecis   0.0  ...       5.740975   
110040        1100403             Alto Paraíso   0.0  ...       8.006356   
110034        1100346         Alvorada D'Oeste   0.0  ...       7.524749   
110002        1100023                Ariquemes   0.0  ...      41.231349   
...               ...                      ...   ...  ...            ...   
522200        5222005               Vianópolis   0.0  ...       5.983062   
522205        5222054           Vicentinópolis   0.0  ...       3.554695   
522220        5222203                 Vila Boa   0.0  ...       2.214672   
522230        5222302            Vila Propício   0.0  ...       2.496641   
530010        5300108                 Brasília   0.0  ...    1191.510567   

         expected_2017  expected_2018  expected_2019  SIR_2014  SIR_2015  \
codmuni                                                                    
110001       20.593672       7.461287      12.643632       0.0       0.0   
110037       10.734248       3.871777       6.517421       0.0       0.0   
110040       14.952320       5.401550       9.109526       0.0       0.0   
110034       14.007107       5.086903       8.664281       0.0       0.0   
110002       77.271117      27.668192      46.704364       0.0       0.0   
...                ...            ...            ...       ...       ...   
522200       11.107703       4.046008       6.952129       0.0       0.0   
522205        6.609486       2.394146       4.079875       0.0       0.0   
522220        4.141425       1.468607       2.462791       0.0       0.0   
522230        4.618760       1.709203       2.933614       0.0       0.0   
530010     2228.028801     801.517853    1362.688808       0.0       0.0   

         SIR_2016  SIR_2017  SIR_2018  SIR_2019  
codmuni                                          
110001   0.000000  0.000000  0.000000  0.000000  
110037 

In [330]:
final[2015].sum()

670.0

In [329]:
data.year.value_counts()

2017.0    166345
2019.0    104125
2016.0     89531
2018.0     60587
2015.0       670
0.0           15
2014.0        10
Name: year, dtype: int64

In [331]:
final.to_excel('standard_incidence_ratio_chik.xls')